docs/architecture.rst

```diff
@@ -0,0 +1,57 @@
+Architecture
+============
+
+.. note::
+
+   This page describes technical details about EXtra-data. You shouldn't need
+   this information to use it.
+
+Objects
+-------
+
+The :class:`.DataCollection` class is the central piece of EXtra-data. It
+represents a collection of XFEL data sources and their keys, for a set of train
+IDs. It refers to data in one or more files (a run directory is often the
+starting point). A subset of its sources/keys or train IDs may be selected to
+make a new, more restricted :class:`.DataCollection`.
+
+:class:`.KeyData` represents data for a single source & key, selected from a
+``DataCollection`` like ``run[source, key]``. This data may still be spread
+across several files. The data can be loaded into a NumPy array, among other
+types.
+
+:class:`.FileAccess` manages access to a single EuXFEL format HDF5 file,
+including caching index information. There should only be one ``FileAccess``
+object per file on disk, even if multiple ``DataCollection`` and ``KeyData``
+objects refer to it.
+
+Modules
+-------
+
+- ``exceptions`` defines some custom error classes.
+- ``file_access`` contains :class:`.FileAccess` (described above), along with
+  machinery to keep the number of open files under a limit.
+- ``locality`` can check whether files are available on disk or on tape,
+  and is specific to EuXFEL & DESY infrastructure.
```

Comment:
It should work with any dCache, not only with DESY or EuXFEL

Reply:
Thanks! I thought we were checking for specific paths as part of it, but I was wrong. I've corrected that.